spec/models/bottleneck_event_spec.rb

```diff
@@ -76,4 +76,37 @@
       expect(query).to match(/resource_type = 'EmsCluster'.*resource_type = 'Host'/)
     end
   end
+
+  describe ".last_created_on" do
+    context "returns the last created_on date" do
+      def expect_last_created_on_to_eq_resource_last_created_on(resource)
+        bottleneck_event = BottleneckEvent.create!(:resource => resource)
+        expect(described_class.last_created_on(resource)).to eq(bottleneck_event.reload.created_on)
+      end
```

Comment:
(Since you asked me on Gitter...)

I think this is perfectly fine by any means, but I do wonder if you could utilize helpers and contexts more, maybe something like...

```
    describe "#last_created_on" do
      subject { described_class.last_created_on(resource) }
      let!(:bottleneck_event) { BottleneckEvent.create!(:resource => resource) }
      let(:resource) { FactoryGirl.create(resource_name) }

      context "for a host_redhat resource" do
        let(:resource_name) { :host_redhat }
        it { is_expected.to be_same_time_as(bottleneck_event.reload.created_on) }
      end

      context "for a host_vmware resource" do
        let(:resource_name) { :host_vmware }
        it { is_expected.to be_same_time_as(bottleneck_event.reload.created_on) }
      end

      context "for a miq_enterprise resource" do
        let(:resource_name) { :miq_enterprise }
        it { is_expected.to be_same_time_as(bottleneck_event.reload.created_on) }
      end

      context "for a ems_redhat resource" do
        let(:resource_name) { :ems_redhat }
        it { is_expected.to be_same_time_as(bottleneck_event.reload.created_on) }
      end

      context "for a ems_cluster_openstack resource" do
        let(:resource_name) { :ems_cluster }
        it { is_expected.to be_same_time_as(bottleneck_event.reload.created_on) }
      end
    end
```

*Includes `be_same_time_as`